File: src/GFATeamManager.Domain/Entities/PreRegistration.cs
using GFATeamManager.Domain.Enums;

namespace GFATeamManager.Domain.Entities;

public class PreRegistration : BaseEntity
{
    public string Cpf { get; set; } = string.Empty;
    public ProfileType Profile { get; set; }
    public PlayerUnit? Unit { get; set; }
    public PlayerPosition? Position { get; set; }
    public string ActivationCode { get; set; }
    public DateTime ExpirationDate { get; set; }
    public bool IsUsed { get; set; }
    public DateTime? UsedAt { get; set; }
    public Guid? UserId { get; set; }
    public User? User { get; set; }
    
    public PreRegistration()
    {
        ActivationCode = GenerateRandomCode();
        ExpirationDate = DateTime.UtcNow.AddDays(7);
        IsUsed = false;
    }
    
    public bool IsExpired() => DateTime.UtcNow > ExpirationDate;

    public bool CanBeUsed() => !IsUsed && !IsExpired();

    private static string GenerateRandomCode()
    {
        const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
        var random = new Random();
        return new string(Enumerable.Repeat(chars, 8)
            .Select(s => s[random.Next(s.Length)]).ToArray());
    }
}

File: frontend/types/pre-registration.ts
export interface PreRegistration {
  id: string;
  cpf: string;
  profile: number;
  unit?: number;
  position?: number;
  activationCode: string;
  expiresAt: string;
  isUsed: boolean;
  createdAt: string;
}

export interface CreatePreRegistrationRequest {
  cpf: string;
  profile: number;
  unit?: number;
  position?: number;
}

export interface PreRegistrationResponse {
  isSuccess: boolean;
  data: PreRegistration | null;
  errors: string[];
}

export interface PreRegistrationListResponse {
  isSuccess: boolean;
  data: PreRegistration[] | null;
  errors: string[];
}
